pkg/controller/storagecluster/placement.go

```diff
@@ -51,25 +55,39 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 	return placement
 }
 
-//convertLabelToNodeSelector returns NodeSelectorTerm type from a given LabelSelector
-func convertLabelToNodeSelector(labelSelector metav1.LabelSelector) corev1.NodeSelectorTerm {
-	term := corev1.NodeSelectorTerm{}
+//convertLabelToNodeSelectorRequirements returns a NodeSelectorRequirement list from a given LabelSelector
+func convertLabelToNodeSelectorRequirements(labelSelector metav1.LabelSelector) []corev1.NodeSelectorRequirement {
+	reqs := []corev1.NodeSelectorRequirement{}
 	for key, value := range labelSelector.MatchLabels {
 		req := corev1.NodeSelectorRequirement{}
 		req.Key = key
 		req.Operator = corev1.NodeSelectorOpIn
 		req.Values = append(req.Values, value)
-		term.MatchExpressions = append(term.MatchExpressions, req)
+		reqs = append(reqs, req)
 	}
 	numIter := len(labelSelector.MatchExpressions)
 	for i := 0; i < numIter; i++ {
 		req := corev1.NodeSelectorRequirement{}
 		req.Key = labelSelector.MatchExpressions[i].Key
 		req.Operator = corev1.NodeSelectorOperator(labelSelector.MatchExpressions[i].Operator)
 		req.Values = labelSelector.MatchExpressions[i].Values
-		term.MatchExpressions = append(term.MatchExpressions, req)
+		reqs = append(reqs, req)
+	}
+	return reqs
+}
+
+func appendNodeRequirements(placement *rookv1.Placement, reqs ...corev1.NodeSelectorRequirement) {
```

Comment:
This seems like an enhancement which isn't related to LabelSelectors. Should be in it's own commit.

Reply:
This very function fixes the actual segfault being reported. Because the same code was in two places, the other segfult would have been hit once the original one had been cleared. Making it a common function fixes both spots.